src/Model/Item.php

```diff
@@ -839,6 +839,81 @@ public static function insert($item, $force_parent = false, $notify = false, $do
 		return $current_post;
 	}
 
+	/**
+	 * @brief Distributes public items to the receivers
+	 *
+	 * @param integer $itemid Item ID that should be added
+	 */
+	public static function distribute($itemid)
+	{
+		$condition = ["`id` IN (SELECT `parent` FROM `item` WHERE `id` = ?)", $itemid];
+		$parent = dba::selectFirst('item', ['owner-id'], $condition);
+		if (!DBM::is_result($parent)) {
+			return;
+		}
+
+		// Only distribute public items from native networks
+		$condition = ['id' => $itemid, 'uid' => 0,
+			'network' => [NETWORK_DFRN, NETWORK_DIASPORA, NETWORK_OSTATUS, ""],
+			'visible' => true, 'deleted' => false, 'moderated' => false, 'private' => false];
+		$item = dba::selectFirst('item', [], ['id' => $itemid]);
+		if (!DBM::is_result($item)) {
+			return;
+		}
+
+		unset($item['id']);
+
+		$condition = ["`nurl` IN (SELECT `nurl` FROM `contact` WHERE `id` = ?) AND `uid` != 0 AND NOT `blocked` AND `rel` IN (?, ?)",
+			$parent['owner-id'], CONTACT_IS_SHARING,  CONTACT_IS_FRIEND];
+		$contacts = dba::select('contact', [], $condition);
+		while ($contact = dba::fetch($contacts)) {
+			self::storeForUser($itemid, $item, $contact);
+		}
+	}
+
+	/**
+	 * @brief Store public items for the receivers
+	 *
+	 * @param integer $itemid  Item ID that should be added
+	 * @param array   $item    The item entry that will be stored
+	 * @param array   $contact The contact that will receive the item entry
+	 */
+	private static function storeForUser($itemid, $item, $contact)
```

Comment:
This method doesn’t references the `user` table, shouldn’t it be named `storeForConact` instead?

Reply:
It uses the contact table to find the user.

Reply:
I clarified the whole stuff. The contact array wasn't needed - only the uid.